src/lib/bootloader/proposal_client.rb

```diff
@@ -131,13 +131,15 @@ def description
 
     # make proposal without handling of exceptions
     def make_proposal_raising(attrs)
+      force_reset = attrs["force_reset"]
+      storage_read = Yast::BootStorage.storage_read?
+      storage_changed = Yast::BootStorage.storage_changed?
```

Comment:
please add note to code about that order and side effects. Otherwise in future it will be again accidentally broken.